### DISCUSSION TITLE
Introduce `SocialLinksConfiguration` class to represent the configuration of the social links data editor

### DISCUSSION THREAD

**Discussion by abjerner** [leekelleher/umbraco-contentment#418] [Idea]
Hi @leekelleher 

For an Umbraco 13 project I'm working on, I'm creating and updating data types via C#.

One of them is the social links data editor, and I realized that setting the configuration is not all that friendly as it isn't backed by a C# type. So as you do in those situations, I ended up creating my own classes, and then converting it to the `Dictionary<string, object?>` that Umbraco expects.

With a bit of method chaining, that would then look something like this:

```csharp
var config = SocialLinksConfiguration
    .Create()
    .AddFacebook()
    .AddLinkedIn()
    .AddInstagram()
    .AddTwitter()
    .AddNetwork("umbfyi", "umb.fyi", "https://umb.fyi/", "icon-umbfyi", "#1B264F", "#fff")
    .ConfirmRemoval()
    .MaxItems(17)
    .EnableDevMode();
```

There would also be quivalent properties to initialize the configuration "the normal way".

Would you be interested in a PR for adding something like this directly to the package? Seeing as this would be a breaking change for current versions of Contentment, I had a quick play with this for v6, which made me question how to even set the configuration on an `IDataType` in Umbraco 14 (and whether the `SocialLinksConfiguration` class even makes sense).

`IDataType` now has the `ConfigurationData` and `ConfigurationObject` - which are readonly. `DataType` (the class) has a `SetConfigurationData` method, which let's you set the configuration from a dictionary, but not something similar for setting the configuration of an object. Am I missing something here? It used to be pretty simple in U13 and earlier.

**Comment by leekelleher** [comment]
@abjerner Thank you for the suggestion, I'll give it some thought.

In the meantime, I'm curious what the breaking-change be?

As for v14+, I haven't tried out the `IDataType` configuration (as a 3rd-party extension/consumer), so unsure what to say about it being readonly, but I'll take a look soon.

**Reply by abjerner** [reply]
Could be that people aren't really using this, but the configuration for the data type currently comes back as `IDictionary<string, object?>` as its not backed my a C# type.

Changing this to `SocialLinksConfiguration` or something else would be a breaking change IMO. But if you're okay with this for V5, I'd be happy to submit a PR 😎

**Reply by abjerner** [reply]
@leekelleher would introducing a `SocialLinksConfiguration` for a strongly typed configuration be acceptable for V5 (I'm still on U13 ... because LTS). Question is whether this should be considered a breaking change (see my last comment). Could also be a PR for V6 or a new major (or major minor) version instead.
